hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/ToRowJsonFunction.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.expression;
+
+import com.hazelcast.core.HazelcastJsonValue;
+import com.hazelcast.jet.impl.util.ReflectionUtils;
+import com.hazelcast.jet.sql.impl.JetSqlSerializerHook;
+import com.hazelcast.jet.sql.impl.expression.json.JsonCreationUtil;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.nio.serialization.genericrecord.GenericRecord;
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.SqlErrorCode;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.expression.UniExpressionWithType;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.type.QueryDataType;
+
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Collections.newSetFromMap;
+
+public class ToRowJsonFunction extends UniExpressionWithType<HazelcastJsonValue> implements IdentifiedDataSerializable {
```

Comment:
We need to investigate the option of using `CAST(customTypeField AS JSON)` first rather than having a non-standard function.

Reply:
Unfortunately the converters are in the core module (which doesn't have automated POJO to JSON libraries, only builders afaik), so its not very feasible to use the built in converters mechanism (not without building logic for this anyway), there's a workaround though - replacing the `CAST(obj AS JSON)` with `TO_ROW_JSON` calls.